api/include/opentelemetry/trace/tracer_provider.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/trace/tracer.h"
+
+namespace opentelemetry
+{
+namespace trace
+{
+class TracerProvider
+{
+public:
+  virtual ~TracerProvider()                                                = default;
+  virtual Tracer *const GetTracer(nostd::string_view library_name,
+                                  nostd::string_view library_version = "") = 0;
```

Comment:
There's a lifetime issue with returning a raw pointer like this.

If the TracerProvider destructs, then the Tracer ptrs will no longer be valid.

Please use a type-erased shared_ptr.